api/conf/settings/databases.py

```diff
@@ -12,6 +12,8 @@
         "USER": config("DJANGO_DATABASE_USER", default="deploy"),
         "PASSWORD": config("DJANGO_DATABASE_PASSWORD", default="deploy"),
         "NAME": config("DJANGO_DATABASE_NAME", default="openledger"),
+        # Default of 5 matches RDS documentation's advised max DNS caching time
```

Comment:
Could you help me understand using this DNS caching time to determine how long we should leave open DB connections? My initial suspicion would be that this number could be a lot higher and I'm not seeing how the DNS cache time relates.

Reply:
I'm curious too. Is this in seconds?

Reply:
It is in seconds, @krysal 👍

Reply:
This probably comes down to my lack of understanding of TCP/IP. If the IP address of one of the ends of the connection changes, would the connection break? The reason RDS says to only cache DNS requests to 5 seconds is because the IP address of the database can change (I assume with some frequency and overlap that 5 seconds wouldn't cause an issue). I made an assumption that this would have an effect on long connections, but I could be wrong about that. Do you know for certain either way?

We could try it at a higher number and reduce it, if you like. It's just an environment variable though so we don't necessarily need to decide that at this point and can change it as we please in production using the environment variable.

Reply:
Did a quick search around and confirmed that my assumption is correct. A [TCP connection cannot be maintained if the IP of either side changes during the connection](https://networkengineering.stackexchange.com/questions/12565/how-is-a-tcp-connection-maintained-over-a-changing-ip-address). I think this implies that we cannot reliably keep connections alive for longer than RDS recommends caching the DNS response.

Reply:
Ah, I see. You're thinking is that matching the DNS cache time will ensure that connections aren't left open, resolving to an IP address that's no longer valid, like in the case of a failover or any other reason the database IP might change. I hadn't considered that.

Where are you seeing the 5 second documentation? In this ["Amazon RDS basic operational guidelines"](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_BestPractices.html) guide I _think_ I see 30 seconds recommended:

> If your client application is caching the Domain Name Service (DNS) data of your DB instances, set a time-to-live (TTL) value of less than 30 seconds. The underlying IP address of a DB instance can change after a failover. Caching the DNS data for an extended time can thus lead to connection failures. Your application might try to connect to an IP address that's no longer in service.

One more thing. I see a new-ish Django setting added in 4.1 which will run a healthcheck on persistient connections before reusing them https://docs.djangoproject.com/en/4.2/ref/settings/#conn-health-checks. Perhaps that + a `CONN_MAX_AGE` of 60 would be reasonable? 

I'm also content to start with 5 seconds and experiment with larger values in the future. Thanks for explaining 😄

Reply:
I must have been reading https://docs.aws.amazon.com/whitepapers/latest/amazon-aurora-mysql-db-admin-handbook/dns-caching.html which is Aurora specific, I now realise. We can raise this to 30 seconds, good catch.

`CONN_HEALTH_CHECKS` looks good to enable as well, will do!

Reply:
Awesome, PR approved. I learned a lot here about DNS caching, thanks again for your patience!